src/syntax-tree/module.ts

```diff
@@ -563,7 +563,15 @@ export class Module {
                     this.addHighlightToConstruct(curr, ERROR_HIGHLIGHT_COLOUR);
                 }
             } else if (curr instanceof Expression && curr.tokens.length > 0) {
-                Q.push(...curr.tokens);
+                const addHighlight = curr instanceof ListLiteralExpression && !curr.isHolePlacementValid();
+                ret = addHighlight ? CodeStatus.ContainsEmptyHoles : ret;
+
+                for (let i = 0; i < curr.tokens.length; i++) {
+                    if (curr.tokens[i] instanceof TypedEmptyExpr && addHighlight && i < curr.tokens.length - 2) {
```

Comment:
If the hole placement is not valid all holes, but the last one should be highlighted.